src/plugins/lanplus/lanplus.c

```diff
@@ -785,9 +785,7 @@ ipmi_lan_poll_single(struct ipmi_intf * intf)
 			if (extra_data_length > 0) {
 				rsp->data_len = extra_data_length;
 				memmove(rsp->data, rsp->data + offset, extra_data_length);
-				offset = 0;
-				payload_start = 0;
-				payload_size = extra_data_length;
+				memset(rsp->data + rsp->data_len, 0, IPMI_BUF_SIZE - rsp->data_len);
```

Comment:
First of all, please use `sizeof(rsp->data)` instead of `IPMI_BUF_SIZE`. Never copy the size of any object from it's definition, always use `sizeof()`. Then whoever updates the definition later won't need to also update a million of other places.

Second, it looks like this change is not correct. It may need to go before `memmove()`. Or maybe it needs to go inside `read_ipmi_response()` instead or again into `sendrecv()` ?

Reply:
I agree sizeof() is better than IPMI_BUF_SIZE. The only reason I used IPMI_BUF_SIZE is to be consistent with other drivers (e.g. Line 655 of src/lan/lan.c, end of function ipmi_lan_poll_recv()).  Please confirm you do want this to be sizeof() and I'll update the change. I don't plan to update other drivers which do not have issues.

memset() position is correct since it sets only the rest of the buffer to \0, which is actually doing the function as you mentioned in your comments for Line 1577 of lib/ipmi_hpmfwupg.c above. Moving it before memmove() will corrupt the received data before moving it to the beginning of the buffer. Please refer to the same code I just mentioned.

Reply:
I do want it to be `sizeof()`. There is plenty of less-than-optimal legacy code. Unless explicitly stated in the Coding Style, please don't blindly follow the pre-existing patterns.Thank you.